src/vs/workbench/contrib/terminal/browser/widgets/terminalQuickFixWidget.ts

```diff
@@ -0,0 +1,265 @@
+						return null;
+					},
+					getWidgetAriaLabel: () => localize({ key: 'customQuickFixWidget', comment: ['A terminal quick fix Option'] }, "Terminal Quick Fix Widget"),
+					getRole: () => 'option',
+					getWidgetRole: () => 'terminal-quickfix-widget'
+				},
+			}
+		}, fixes, showHeaders, previewSelectedTerminalQuickFixCommand, acceptSelectedTerminalQuickFixCommand, (element: ListMenuItem<TerminalQuickFix>) => { return element.kind !== 'header' && !element.item?.disabled; }, onDidSelect, contextViewService);
+	}
+
+	public toMenuItems(inputActions: readonly TerminalQuickFix[], showHeaders: boolean): ListMenuItem<TerminalQuickFix>[] {
+		const menuItems: TerminalQuickFixListItem[] = [];
+		menuItems.push({
+			kind: ActionListItemKind.Header,
+			group: {
+				kind: CodeActionKind.QuickFix,
+				title: 'Quick fix...',
+				icon: { codicon: Codicon.lightBulb }
+			}
+		});
+		for (const action of showHeaders ? inputActions : inputActions.filter(i => !!i.action)) {
+			if (!action.disabled && action.action) {
+				menuItems.push({
+					kind: ActionListItemKind.Action,
+					item: action,
+					group: {
+						kind: CodeActionKind.QuickFix,
+						icon: { codicon: action.action.id === 'quickFix.opener' ? Codicon.link : Codicon.run },
+						title: action.action!.label
+					},
+					disabled: false,
+					label: action.title
+				});
+			}
+		}
+		return menuItems;
+	}
+}
+
+interface TerminalQuickFixListItem extends ListMenuItem<TerminalQuickFix> {
+	readonly item?: TerminalQuickFix;
+}
+
+registerAction2(class extends Action2 {
+	constructor() {
+		super({
+			id: 'hideTerminalQuickFixWidget',
+			title: {
+				value: localize('hideTerminalQuickFixWidget.title', "Hide terminal quick fix widget"),
+				original: 'Hide terminal quick fix widget'
+			},
+			precondition: Context.Visible,
+			keybinding: {
+				weight,
+				primary: KeyCode.Escape,
+				secondary: [KeyMod.Shift | KeyCode.Escape]
+			},
+		});
+	}
+
+	run(): void {
+		TerminalQuickFixWidget.INSTANCE?.hide();
+	}
+});
+
+registerAction2(class extends Action2 {
+	constructor() {
+		super({
+			id: 'selectPrevTerminalQuickFix',
+			title: {
+				value: localize('selectPrevTerminalQuickFix.title', "Select previous terminal quick fix"),
+				original: 'Select previous terminal quick fix'
+			},
+			precondition: Context.Visible,
+			keybinding: {
+				weight,
+				primary: KeyCode.UpArrow,
+				secondary: [KeyMod.CtrlCmd | KeyCode.UpArrow],
+				mac: { primary: KeyCode.UpArrow, secondary: [KeyMod.CtrlCmd | KeyCode.UpArrow, KeyMod.WinCtrl | KeyCode.KeyP] },
+			}
+		});
+	}
+
+	run(): void {
+		TerminalQuickFixWidget.INSTANCE?.focusPrevious();
+	}
+});
+
+registerAction2(class extends Action2 {
+	constructor() {
+		super({
+			id: 'selectNextTerminalQuickFix',
+			title: {
+				value: localize('selectNextTerminalQuickFix.title', "Select next terminal quick fix"),
+				original: 'Select next terminal quick fix'
+			},
+			precondition: Context.Visible,
+			keybinding: {
+				weight,
```

Comment:
For a ux perspective, If we're reusing the code action widget, shouldn't be be able to reuse the same keybindings? Are there cases where the terminal code action widget needs custom keybindings or behavior?

Reply:
The user may want a custom keybinding for the terminal but I guess you could do that via context keys like `terminalFocus`. Where would this go if it's shared between editor and workbench?

Reply:
couldn't find the right place for this, see the commit below where I tried and got close